### PR TITLE
ref(ts): Convert versionHoverCard

### PR DESCRIPTION
**Pull request by leedongwei**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/versionHoverCard.tsx [hunk 1]
```diff
@@ -106,10 +117,13 @@ class VersionHoverCard extends React.Component {
...
   getBody() {
+    const {releaseVersion} = this.props;
     const {release, deploys} = this.state;
-    const {version} = this.props;
-    const lastCommit = release.lastCommit;
+    if (!release) {
+      return {header: null, body: null};
+    }
 
+    const lastCommit = release.lastCommit;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/versionHoverCard.tsx:126]: Can we destructure here?

**Review by billyvg — APPROVED** [review]
LGTM, is this used in getsentry?

**Comment by leedongwei** [comment]
Good point, will check in `getsentry` before merging.

File: src/sentry/static/sentry/app/components/versionHoverCard.tsx [hunk 2]
```diff
@@ -33,31 +43,31 @@ class VersionHoverCard extends React.Component {
     hasRepos: false,
     deploys: [],
     release: null,
-  };
+  } as State;
```
Comment (scttcper) [inline src/sentry/static/sentry/app/components/versionHoverCard.tsx:46]: should this be `state: State = {}`?

File: src/sentry/static/sentry/app/components/versionHoverCard.tsx [hunk 3]
```diff
@@ -16,15 +15,26 @@ import withApi from 'app/utils/withApi';
...
+type Props = {
+  api: Client;
+  orgSlug: string;
+  projectSlug: string;
+  releaseVersion: string;
+};
+type State = {
+  loading: boolean;
+  error: boolean;
+  data: {};
```
Comment (scttcper) [inline src/sentry/static/sentry/app/components/versionHoverCard.tsx:30]: Is data used?

File: src/sentry/static/sentry/app/components/versionHoverCard.tsx (hunk 2 above)
Comment (leedongwei) [reply src/sentry/static/sentry/app/components/versionHoverCard.tsx]: Good catch

File: src/sentry/static/sentry/app/components/versionHoverCard.tsx (hunk 3 above)
Comment (leedongwei) [reply src/sentry/static/sentry/app/components/versionHoverCard.tsx:30]: This component is being reworked in another PR and `data` is being deprecated.

**Comment by leedongwei** [comment]
Force-push to rebase on `master`

**Review by scttcper — APPROVED** [review]

**Pull request merged**